packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3445,6 +3423,33 @@ export class ContainerRuntime
 				};
 			},
 		);
+
+		// We choose to close the summarizer after the snapshot cache is updated to avoid
+		// situations which the main client (which is likely to be re-elected as the leader again)
+		// loads the summarizer from cache.
+		if (this.summaryStateUpdateMethod === "restart") {
```

Comment:
should we also consider not going into the restart if we get a specific versionid ? remmeber the scenario for backward compatibility ?

Reply:
So my understanding is that back-compat wise, this versionId comes from the latest ack that we receive. Potentially what would happen would be multiple versionId calls. The problem is that the refresh code is buggy and we are better of restarting the summarizer a few times in terms of reliability.